packages/rpc-graphql/src/resolvers/account.ts

```diff
@@ -0,0 +1,89 @@
+import { SolanaRpcMethods } from '@solana/rpc-core';
+import { Rpc } from '@solana/rpc-transport/dist/types/json-rpc-types';
+import { GraphQLResolveInfo } from 'graphql';
+
+import { GraphQLCache } from '../cache';
+import { AccountQueryArgs } from '../schema/account';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function refineJsonParsedAccountData(jsonParsedData: any) {
+    const meta = {
+        program: jsonParsedData.program,
+        space: jsonParsedData.space,
+        type: jsonParsedData.parsed.type,
+    };
+    const data = jsonParsedData.parsed.info;
+    return { data, meta };
+}
+
+// Default to jsonParsed encoding if none is provided
+export async function resolveAccount(
+    { address, encoding = 'jsonParsed', ...config }: AccountQueryArgs,
+    cache: GraphQLCache,
+    rpc: Rpc<SolanaRpcMethods>,
+    info?: GraphQLResolveInfo
+) {
+    // If a user only requests the account's address, don't call the RPC
+    if (info && info.fieldNodes[0].selectionSet) {
+        const selectionSet = info.fieldNodes[0].selectionSet;
+        const requestedFields = selectionSet.selections.map(field => {
+            if (field.kind === 'Field') {
+                return field.name.value;
+            }
+            return null;
+        });
+        if (requestedFields && requestedFields.length === 1 && requestedFields[0] === 'address') {
+            return { address };
+        }
+    }
+
+    const requestConfig = { encoding, ...config };
+
+    const cached = cache.get(address, requestConfig);
+    if (cached !== null) {
+        return cached;
+    }
+
+    const account = await rpc
+        .getAccountInfo(address, requestConfig as Parameters<SolanaRpcMethods['getAccountInfo']>[1])
+        .send()
+        .then(res => res.value)
+        .catch(e => {
+            throw e;
+        });
```

Comment:
Oh, I've gotta get you on the `DataLoader` train ASAP.

In particular, the architecture you've developed here won't actually dedupe requests for the same thing when multiple resolvers are working in parallel, because you don't _save_ the cache result until the request is over. All of them will kick off, think there's no cache, make _n_ queries, and then save the same (maybe) result in the cache at the end.

DataLoader:

1. Specify all of the resolvers in the form of dataloaders; bind the rpc at the very very top level context
2. Pass those dataloaders down to resolvers through context
3. Call `.load()` on them.

https://github.com/steveluscher/zero-to-graphql/blob/3daf2d8dc2ac9bc63cf16966d6489fe43cbddbe9/zero-node/index.js#L32-L48

Reply:
@buffalojoec, to get you started: https://app.graphite.dev/github/pr/solana-labs/solana-web3.js/1827/refactor-experimental-use-DataLoader-to-coalesce-multiple-requests-for-the-same-block#

Reply:
Love it. Thanks!